Tests/SSL/test_ssl_containers.py

```diff
@@ -622,3 +624,45 @@ def test_simclr_dataset_length(test_output_dirs: OutputFolderForTests,
         val_loaders = container.get_data_module().val_dataloader()
         assert isinstance(val_loaders, CombinedLoader)
         assert len(val_loaders) == expected_num_val_iters
+
+
+def test_simclr_dataloader_type() -> None:
+    """ This test checks if the transform pipeline of a SSL job can handle different
+    data types coming from the dataloader.
+    """
+    def check_types_in_batch(batch: Dict) -> None:
+        assert type(batch[SSLDataModuleType.ENCODER][0][0]) == torch.Tensor
+        assert type(batch[SSLDataModuleType.ENCODER][0][1]) == torch.Tensor
+        assert type(batch[SSLDataModuleType.ENCODER][1]) == torch.Tensor
+        assert type(batch[SSLDataModuleType.LINEAR_HEAD][0]) == torch.Tensor
+        assert type(batch[SSLDataModuleType.LINEAR_HEAD][1]) == torch.Tensor
+        assert type(batch[SSLDataModuleType.LINEAR_HEAD][2]) == torch.Tensor
```

Comment:
Here it would be safer to use `isinstance(...)`, in case we get a `Tensor` subclass, for example.